scalyr_agent/log_processing.py

```diff
@@ -2034,6 +2034,10 @@ def __init__(
         # happens).  However, this is how the old agent worked and the UI relies on it, so we just keep the old system
         # going for now.
         self.__thread_name = "Lines for file %s" % file_path
+
+        # Note: thread id is also used as a unique identifier for LogFileProcessors (see #CT-107)
+        # So even if "thread_id" is no longer used by the Scalyr API, we still need it as a unique
+        # integer identifier for the LogFileProcessor
         self.__thread_id = LogFileProcessor.generate_unique_id()
```

Comment:
I had a look and it looks like this returns a monotonically increasing integer which should be fine.

If we ever needed to avoid the need to use a lock we could probably also use uuid v4, but since we already have that locking code in place, this is fine as well.

Reply:
Yeah, Imron and I had that exact same mini-debate.  We agreed this is good enough for now.